cmd/ack-generate/command/apis.go

```diff
@@ -84,138 +83,36 @@ func generateAPIs(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("service %s not found", svcAlias)
 		}
 	}
-	if !optDryRun {
-		apisVersionPath = filepath.Join(optAPIsOutputPath, svcAlias, "apis", optGenVersion)
-		if _, err := ensureDir(apisVersionPath); err != nil {
-			return err
-		}
-	}
 	g, err := generate.New(
-		sdkAPI, optGenVersion, optGeneratorConfigPath, optTemplatesDir, config.Default,
+		sdkAPI, optGenVersion, optGeneratorConfigPath, ackgenerate.DefaultConfig,
 	)
 	if err != nil {
 		return err
 	}
-
-	crds, err := g.GetCRDs()
-	if err != nil {
-		return err
-	}
-	typeDefs, _, err := g.GetTypeDefs()
-	if err != nil {
-		return err
-	}
-	enumDefs, err := g.GetEnumDefs()
+	ts, err := ackgenerate.APIs(g, optTemplatesDir)
 	if err != nil {
 		return err
 	}
 
-	if err = writeDocGo(g); err != nil {
+	if err = ts.Execute(); err != nil {
 		return err
 	}
 
-	if err = writeGroupVersionInfoGo(g); err != nil {
-		return err
-	}
-
-	if err = writeEnumsGo(g, enumDefs); err != nil {
-		return err
-	}
-
-	if err = writeTypesGo(g, typeDefs); err != nil {
-		return err
-	}
-
-	for _, crd := range crds {
-		if err = writeCRDGo(g, crd); err != nil {
+	apisVersionPath = filepath.Join(optAPIsOutputPath, svcAlias, "apis", optGenVersion)
+	for path, contents := range ts.Executed() {
+		if optDryRun {
+			fmt.Printf("============================= %s ======================================", path)
```

Comment:
nit: all the --dry-run like commands needs a new ligne after printing headers. Thinking something like : `fmt.Printf("============================= %s ======================================\n", path)`

Reply:
Nice catch. Fixed.